bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/utils/DbMetaData.java

```diff
@@ -0,0 +1,117 @@
+package org.openhab.persistence.jdbc.utils;
+
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+
+import org.knowm.yank.Yank;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+/**
+ *
+ * @author Helmut Lehmeyer
+ * @since 1.8.0
+ */
+public class DbMetaData {
+
+    private static final Logger logger = LoggerFactory.getLogger(DbMetaData.class);
+
+    private int dbMajorVersion;
+    private int dbMinorVersion;
+    private int driverMajorVersion;
+    private int driverMinorVersion;
+    private String dbProductName;
+    private String dbProductVersion;
+
+    public DbMetaData() {
+        HikariDataSource h = Yank.getDefaultConnectionPool();
+        // HikariDataSource h = Yank.getDataSource();
+
+        DatabaseMetaData meta;
+        try {
+            meta = h.getConnection().getMetaData();
+
+            // Oracle (and some other vendors) do not support
+            // some the following methods; therefore, we need
+            // to use try-catch block.
+            try {
+                dbMajorVersion = meta.getDatabaseMajorVersion();
+                logger.debug("dbMajorVersion = '{}'", dbMajorVersion);
+            } catch (Exception e) {
+                logger.error("Asking for 'dbMajorVersion' is a unsupported: '{}'", e);
```

Comment:
Do you mean `e.getMessage()` or dump the stack trace, so no `{}` in message string?